base_layer/core/src/block.rs

```diff
@@ -36,6 +37,15 @@ pub struct Block {
     pub body: AggregateBody,
 }
 
+impl Block {
+    /// This function will check the block to ensure that all UTXO's are validly constructed and that all signatures are
+    /// valid
+    pub fn validate(&self) -> Result<(), TransactionError> {
```

Comment:
This function currently doesn't do what it says on the box AFAICT right?

Reply:
Its suppose to, unless I am reusing the transaction functions wrongly?

Reply:
`validate_internal...` cannot check that inputs are from the valid UTXO set

Reply:
O yes, thats only checked later when we prune all the inputs. At this point we can only assume the inputs are valid unspent UTXO's and that the new UTXO's are validly constructed. 
The checks are meant to be ordered from least expensive to most expensive so we catch out bad blocks as cheaply as possible